spec/tree_visibility_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'spec_helper'
+require './tree_visibility'
+
+describe Tree do
+  describe '#solution' do
+    let(:instance) { described_class.new }
+
+    context 'given example' do
+      it 'should return 4 given [5, [3, 20, 21], [10, 1, nil]]' do
+        tree_l = TreeNode.new(3, TreeNode.new(21), TreeNode.new(20))
+        tree_r = TreeNode.new(10, TreeNode.new(1))
+        tree = TreeNode.new(5, tree_l, tree_r)
```

Comment:
Using `let` maybe? https://relishapp.com/rspec/rspec-core/v/2-11/docs/helper-methods/let-and-let